source/postprocess/visualization/maximum_shear_stress.cc

```diff
@@ -0,0 +1,280 @@
+/*
+  Copyright (C) 2016 - 2017 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file LICENSE.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#include <aspect/postprocess/visualization/maximum_shear_stress.h>
+#include <aspect/gravity_model/interface.h>
+#include <aspect/utilities.h>
+
+
+namespace aspect
+{
+  namespace Postprocess
+  {
+    namespace VisualizationPostprocessors
+    {
+      template <int dim>
+      void
+      MaximumShearStress<dim>::
+      evaluate_vector_field(const DataPostprocessorInputs::Vector<dim> &input_data,
+                            std::vector<Vector<double> > &computed_quantities) const
+      {
+        const unsigned int n_quadrature_points = input_data.solution_values.size();
+        Assert (computed_quantities.size() == n_quadrature_points,    ExcInternalError());
+        Assert ((computed_quantities[0].size() == dim),
+                ExcInternalError());
+        Assert (input_data.solution_values[0].size() == this->introspection().n_components,   ExcInternalError());
+        Assert (input_data.solution_gradients[0].size() == this->introspection().n_components,  ExcInternalError());
+
+        MaterialModel::MaterialModelInputs<dim> in(input_data,
+                                                   this->introspection());
+        MaterialModel::MaterialModelOutputs<dim> out(n_quadrature_points,
+                                                     this->n_compositional_fields());
+
+        // Compute the viscosity...
+        this->get_material_model().evaluate(in, out);
+
+        // ...and use it to compute the stresses and from that the
+        // maximum compressive stress direction
+        for (unsigned int q=0; q<n_quadrature_points; ++q)
+          {
+            const SymmetricTensor<2,dim> strain_rate = in.strain_rate[q];
+            const SymmetricTensor<2,dim> compressible_strain_rate
+              = (this->get_material_model().is_compressible()
+                 ?
+                 strain_rate - 1./3 * trace(strain_rate) * unit_symmetric_tensor<dim>()
+                 :
+                 strain_rate);
+
+            const double eta = out.viscosities[q];
+
+            // first compute the stress tensor, ignoring the pressure
+            // for the moment (the pressure has no effect on the
```

Comment:
does this still apply? in contrast to the place where you copied this from, here you *really* don't care about the pressure, right?